sc-tools/sc-config/src/sc_config.cpp

```diff
@@ -66,14 +67,68 @@ std::string ScConfigGroup::operator[](std::string const & key) const
   if (it == m_pathKeys.cend())
     return value;
 
-  if (value[0] == '/')
-    stream << value;
-  else
-    stream << m_configPath << value;
+  std::string valueStr(value);
+  std::string::size_type start = 0;
+
+  while (true)
+  {
+    std::string::size_type end = valueStr.find(ScConfig::PATHS_SEPARATOR, start);
+    std::string path = valueStr.substr(start, end - start);
+
+    path.erase(std::remove_if(path.begin(), path.end(), ::isspace), path.end());
+
+    if (!path.empty())
+    {
+      stream << NormalizePath(path);
+      stream << (end == std::string::npos ? "" : std::string(1, ScConfig::PATHS_SEPARATOR));
+    }
+
+    if (end == std::string::npos)
+      break;
+
+    start = end + 1;
+  }
 
   return stream.str();
 }
 
+std::string ScConfigGroup::NormalizePath(std::string const & path) const
```

Comment:
on what case is this implementation better then 
```

  if (path.empty())
    return "";

  std::filesystem::path const & currentDir = std::filesystem::current_path();

  std::filesystem::path fullPath =
      (path[0] == ScConfig::SLASH) ? std::filesystem::path(path) : currentDir / std::filesystem::path(m_configPath) / path;

  fullPath = fullPath.lexically_normal();
  return fullPath;

```

Reply:
To reproduce the issue, please follow these steps:
1. ./build_sc_machine.sh
2. cd build
3. ./sc-machine -c ../bin/../sc-machine.ini

You will observe that the paths to the knowledge base binaries, log files, and extensions are absolute. 

This implementation normalizes the paths by stripping them of absolute prefixes. This version organizes the information more clearly and uses consistent formatting for commands and explanations.

Reply:
what's wrong with absolute path? someone who looks through logs will be able to tell what file is used for what and he wouldn't have to know what is currentDir to tell the location of a file

Reply:
also you can take a look into what internet is saying about 'kiss principle'

Reply:
Current directory is the directory from which you run the binary. There is no need to reflect it constantly in the logs?

Reply:
kiss principle is not about something like this

Reply:
> Current directory is the directory from which you run the binary.

Terminals show this directory

Reply:
still, what's wrong with absolute path? kiss principle is about not writing 20 lines of code for something that can be made with 1 line

Reply:
Yes, KISS is about this, but these two code are different, they do not the same. Absolute paths reveal the internal structure of the file system. My implementation hides them, making them available in logs only to the person who calls the binary

Reply:
do this commands starting from root of sc-machine(use your name of the folder with sc-machine, I'll use `main-sc-machine`)
```
cd ..
mkdir test
cd test
mkdir main-sc-machine
cd main-sc-machine
mkdir bin
cd bin
../../../main-sc-machine/bin/sc-machine -c ../../../main-sc-machine/sc-machine.ini
```
and extensions path will be broken

Reply:
fixed